### PR TITLE
add tests for uber applier

### PR DESCRIPTION
**Pull request by CaojiamingAlan**
#16036 
Add tests for uber applier.
Test cases involve building correct applier chains when alarms are on and off.

### CODE REVIEW

File: server/etcdserver/apply/uber_applier_test.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+					{
+						Request: &pb.RequestOp_RequestRange{
+							RequestRange: &pb.RangeRequest{
+								Key: []byte(key),
+							},
+						},
+					},
+				}}},
+			expectError: nil,
+		},
```
Comment (chaochn47) [inline server/etcdserver/apply/uber_applier_test.go:183]: Could you please add an conditional delete in the Txn test case? It was the motivation of PR https://github.com/etcd-io/etcd/pull/13435/

Comment (CaojiamingAlan) [reply server/etcdserver/apply/uber_applier_test.go:183]: Sure. Lemme check the background first.

Comment (CaojiamingAlan) [reply server/etcdserver/apply/uber_applier_test.go:183]: I can add a test case, but tbh I don't think that's the part of function I am trying to cover here......a better place would be the unit test of the [server/storage](https://github.com/etcd-io/etcd/tree/833aabe4cd5061c02cb72d010836c7e57510aab2/server/storage) package. However that package has no test coverage either. So nevermind, I will add it.

**Review by chaochn47 — APPROVED** [review]
LGTM, thanks @CaojiamingAlan !

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks @CaojiamingAlan.

**Review by serathius — APPROVED** [review]

**Review by ahrtr — APPROVED** [review]

**Pull request merged**